### PR TITLE
Let Supervisors and Admins Delete Future Court Dates #3215

### LINKED ISSUE #3215: Let Supervisors and Admins Delete Future Court Dates
**What type(s) of user does this feature affect?**
 - supervisors
 - admins

**Description**
Currently there is no way for anyone to delete a court date. We would like to make it so that Supervisors and Admins can delete future court dates. We would like to make it so that on the screen that appears when one selects a Court Date from the Edit Case Details Screen (see screenshot below), there is a big red button under the currently existing box of information that says "Delete Future Court Date". 

**Screenshots of current behavior, if any**  
Future Court Date Without Delete Option

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by Salanoid**
### What github issue is this PR for, if any?
Resolves #3215 
### What changed, and why?
Supervisors and casa admins can delete court dates.

### How will this affect user permissions?
- Volunteer permissions: nothing
- Supervisor permissions: can delete court dates
- Admin permissions: can delete court dates

### How is this tested? (please write tests!) 💖💪
System test in spec/system/court_dates/edit_spec.rb.

### Screenshots please :)
delete_button_snip

### CODE REVIEW

File: app/controllers/court_dates_controller.rb [hunk 1]
```diff
@@ -53,6 +53,12 @@ def update
     end
   end
 
+  def destroy
+    return unless current_user.casa_admin? || current_user.supervisor?
```
Comment (compwron) [inline app/controllers/court_dates_controller.rb:57]: This should use the policy file

Comment (compwron) [reply app/controllers/court_dates_controller.rb:57]: https://github.com/rubyforgood/casa/blob/main/app/policies/court_date_policy.rb

Comment (compwron) [reply app/controllers/court_dates_controller.rb:57]: i.e. `authorize @court_date` or similar

Comment (Salanoid) [reply app/controllers/court_dates_controller.rb:57]: Thanks for your review. I'll take a look at this later today and comeback.

Comment (Salanoid) [reply app/controllers/court_dates_controller.rb]: I made some changes.

File: app/views/court_dates/show.html.erb [hunk 2]
```diff
@@ -64,3 +64,11 @@ casa_case_path(@casa_case) %></h6>
       class: "btn btn-primary" %>
   </div>
 </div>
+
+<% if current_user.casa_admin? || current_user.supervisor? %>
```
Comment (compwron) [inline app/views/court_dates/show.html.erb:68]: anything checking permissions should be in a policy file `if current_user.casa_admin? || current_user.supervisor?`

**Review by compwron — APPROVED** [review]
Yay tests!
Please use the policy file for permissions and then feel free to merge

**Comment by Salanoid** [comment]
Thank you very much!

I made the changes but I can't do the merge by myself.

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
Nice! :)

**Pull request merged**